src/dmd/typesem.d

```diff
@@ -908,13 +898,6 @@ extern(C++) Type typeSemantic(Type type, const ref Loc loc, Scope* sc)
             Dsymbol s;
             mtype.index.resolve(loc, sc, e, t, s);
 
-            //https://issues.dlang.org/show_bug.cgi?id=15478
-            if (s)
```

Comment:
This check seems to be necessary, but the handling can be replaced by `e = symbolToExp(s, loc, sc, false);`

(Also updated my initial patch suggestion)